tests/nat/conftest.py

```diff
@@ -0,0 +1,227 @@
+                                          "dst_ip": SETUP_CONF[interface_type]["vrf"]["red"]["ip"],
+                                          "second_dst_ip": SETUP_CONF[interface_type]["vrf"]["green"]["ip"],
+                                          "public_ip": public_ip,
+                                          "second_public_ip": second_public_ip,
+                                          "gw": duthost.setup()['ansible_facts']['ansible_Vlan1000']['ipv4']['address'],
+                                          "acl_subnet": SETUP_CONF[interface_type]["acl_subnet"]
+                                          }
+                         }
+    try:
+        # Setup interfaces on PTF container
+        conf_ptf_interfaces(ptfhost, duthost, setup_information, interface_type)
+    except Exception as err:
+        teardown_test_env(duthost, ptfhost, setup_information, interface_type)
+        raise err
+    yield interface_type, setup_information
+    # Remove ptf interfaces
+    conf_ptf_interfaces(ptfhost, duthost, setup_information, interface_type, teardown=True)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def apply_global_nat_config(duthost):
+    """
+    generate global NAT configuration files and deploy them on DUT;
+    after test run cleanup artifacts on DUT
+    :param duthost: DUT host object
+    """
+    # Create temporary directory for NAT templates
+    duthost.command("mkdir -p {}".format(DUT_TMP_DIR))
+    # Initialize variables for NAT global table
+    nat_table_vars = {
+        'nat_admin_mode': NAT_ADMIN_MODE,
+        'global_nat_timeout': GLOBAL_NAT_TIMEOUT,
+        'tcp_timeout': GLOBAL_TCP_NAPT_TIMEOUT,
+        'udp_timeout': GLOBAL_UDP_NAPT_TIMEOUT,
+    }
+    duthost.host.options['variable_manager'].extra_vars.update(nat_table_vars)
+    nat_global_config = 'nat_table_global_{}.json'.format(NAT_ADMIN_MODE)
+    nat_config_path = os.path.join(DUT_TMP_DIR, nat_global_config)
+    duthost.template(src=os.path.join(TEMPLATE_DIR, NAT_GLOBAL_TEMPLATE), dest=nat_config_path)
+    # Apply config file
+    duthost.command('sonic-cfggen -j {} --write-to-db'.format(nat_config_path))
+    yield
+    # Remove temporary folders
+    duthost.command('rm -rf {}'.format(DUT_TMP_DIR))
+    # reload config on teardown
+    config_reload(duthost, config_source='minigraph')
+
+
+@pytest.fixture()
+def enable_nat_config(request, duthost):
+    """
+    enable NAT configuration on teardown
+    :param request: pytest request object
+    :param duthost: DUT host object
+    """
+    yield
+    if request.node.rep_call.failed:
+        duthost.command("config nat feature enable")
+
+
+@pytest.fixture()
+def enable_nat_docker(request, duthost):
+    """
+    enable NAT Docker on teardown
+    :param request: pytest request object
+    :param duthost: DUT host object
+    """
+    yield
+    if request.node.rep_call.failed:
+        exec_command(duthost, ["sudo docker start nat"])
+
```

Comment:
Isnt NAT container running by default ? If we need to restart NAT i think it is better to use systemctl command instead of docker commands

Reply:
If you perform `systemctl cat nat.service` you can see that service configured to perform restart every 30 seconds.
`[Service]
Restart=always
RestartSec=30`
That's why start/stop container was being done using docker commands.